lib/model.js

```diff
@@ -357,9 +358,15 @@ function preprocess(req, res, next) {
 }
 
 function query(key) {
-  return function(val, query) {
-    return query[key](val);
-  };
+  if (key === 'within') {
+    return function(val, query) {
```

Comment:
It's because within function takes multiple arguments. For square area there will be 2 arguments, for polygons more.